test/e2e/util.go

```diff
@@ -3753,3 +3767,29 @@ func CoreDump(dir string) {
 		Logf("Error running cluster/log-dump.sh: %v", err)
 	}
 }
+
+// Run a function over each pod in a namespace.
+// Meant to be called in a loop - where the fn does some assertion logic on a test (i.e. look for string in the pod logs)
+func TryEachPod(c *client.Client, ns, selectorKey, selectorValue string, fn func(api.Pod)) {
+	pods := []api.Pod{}
+	for t := time.Now(); time.Since(t) < podListTimeout; time.Sleep(poll) {
```

Comment:
UPDATE tim is referring specifically to the 3775 line rather than not the waitForPodRunning in namespace methods in test/e2e/utils.go ..... so this particular line we can replace w/ until